### PR TITLE
Feature/sc 196

### PR DESCRIPTION
**Pull request by mclaurentiu**

### CODE REVIEW

**Review by CaptainOfHacks — APPROVED** [review]
- tika_adapter has been remove completely, we don't need this adapter in future?

- new configs look good, some config property need refactoring after this, example EURLEX->EU_CELLAR

File: sem_covid/entrypoints/etl_dags/eurlex_covid19.py [hunk 1]
```diff
@@ -62,192 +62,351 @@
...
+        GROUP BY ?work ?title
+        ORDER BY ?work ?title"""
+
+sources = {
+    "EurLex": {"json": config.EURLEX_TIMELINE_JSON, "query": eurlex_query},
+    "Extended EurLex": {"json": config.EURLEX_EXTENDED_JSON, "query": eurlex_extended_query}
+}
+
+
+def get_transformation_rules(rules: str, search_rule: str = SEARCH_RULE):
```
Comment (costezki) [inline sem_covid/entrypoints/etl_dags/eurlex_covid19.py:384]: this is no longer needed, is it?

File: sem_covid/entrypoints/etl_dags/eurlex_covid19.py [hunk 2]
```diff
@@ -62,192 +62,351 @@
 
 SEARCH_RULE = ".[] | "
```
Comment (costezki) [inline sem_covid/entrypoints/etl_dags/eurlex_covid19.py:63]: is this still relevant?

File: sem_covid/entrypoints/etl_dags/eurlex_covid19.py [hunk 3]
```diff
@@ -62,192 +62,351 @@
...
+
+def get_single_item(query, json_file_name):
+    logger.info('Start retrieving EURLex Covid 19 items')
+    minio = MinioAdapter(config.MINIO_URL, config.MINIO_ACCESS_KEY, config.MINIO_SECRET_KEY, config.EURLEX_BUCKET_NAME)
+    minio.empty_bucket(object_name_prefix=None)
+    minio.empty_bucket(object_name_prefix=RESOURCE_FILE_PREFIX)
+    minio.empty_bucket(object_name_prefix=TIKA_FILE_PREFIX)
 
     response = make_request(query)['results']['bindings']
     transformed_json = compile(get_transformation_rules(transformation)).input(response).all()
```
Comment (costezki) [inline sem_covid/entrypoints/etl_dags/eurlex_covid19.py:404]: Oh! Let's externalise this.

File: sem_covid/entrypoints/etl_dags/eurlex_covid19.py [hunk 4]
```diff
@@ -296,32 +455,32 @@ def download_covid19_items():
...
 
                 logger.info(
-                    f"[{index + 1}/{eurlex_items_count}] Downloading PDF manifestation for {item['title']}")
+                    f"[{index + 1}/{items_count}] Downloading PDF manifestation for {item['title']}")
 
                 pdf_file = filename + '_pdf.zip'
                 if download_file(item, pdf_manifestation, pdf_file, minio):
                     counter['pdf'] += 1
         else:
             logger.exception(f"No manifestation has been found for {item['title']}")
```
Comment (costezki) [inline sem_covid/entrypoints/etl_dags/eurlex_covid19.py:464]: work_id is more useful here, though less informative than the title.

File: sem_covid/entrypoints/etl_dags/eurlex_covid19.py [hunk 5]
```diff
@@ -384,14 +543,32 @@ def upload_processed_documents_to_elasticsearch():
...
-                                       document_body=loads(minio.get_object(obj.object_name).decode('utf-8')))
+                             document_body=loads(minio.get_object(obj.object_name).decode('utf-8')))
             object_count += 1
         except Exception as ex:
             logger.exception(ex)
 
     logger.info(f'Sent {object_count} file(s) to ElasticSearch.')
 
 
+def get_items():
```
Comment (costezki) [inline sem_covid/entrypoints/etl_dags/eurlex_covid19.py:554]: this is where I was asking myself if we could spawn a DAG step (operator) for each of the single items that are being downloaded. Then we have granular control within the DAG over each individual download. 
This was possible by embedding these multiple PythonOperators into an operator group. We have a story about this in JIRA.

**Review by costezki — APPROVED** [review]
We can refactor and write unit tests for these dags after the delivery.

File: sem_covid/entrypoints/etl_dags/eurlex_covid19.py (hunk 1 above)
Comment (mclaurentiu) [reply sem_covid/entrypoints/etl_dags/eurlex_covid19.py:384]: it is, it's the eurlex dag (now we have extracted just for the pwdb dag)

File: sem_covid/entrypoints/etl_dags/eurlex_covid19.py (hunk 2 above)
Comment (mclaurentiu) [reply sem_covid/entrypoints/etl_dags/eurlex_covid19.py:63]: yes, used in the eurlex dag

**Pull request merged**